ccl/changefeedccl: TestAlterChangefeedDropTargetAfterTableDrop failed

ccl/changefeedccl.TestAlterChangefeedDropTargetAfterTableDrop [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19914712?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19914712?buildTab=artifacts#/) on master @ [53814aa1b3621edd03b670361236a4a5f2f63546](https://github.com/cockroachdb/cockroach/commits/53814aa1b3621edd03b670361236a4a5f2f63546):


```
=== RUN   TestAlterChangefeedDropTargetAfterTableDrop
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a152b17b8fdc5b459db71a1944ab6ad5/logTestAlterChangefeedDropTargetAfterTableDrop3488072084
    test_log_scope.go:76: use -show-logs to present logs inline
    alter_changefeed_test.go:506: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a152b17b8fdc5b459db71a1944ab6ad5/logTestAlterChangefeedDropTargetAfterTableDrop3488072084
--- FAIL: TestAlterChangefeedDropTargetAfterTableDrop (47.46s)
=== RUN   TestAlterChangefeedDropTargetAfterTableDrop/kafka
    helpers_test.go:1235: making server as secondary tenant
    helpers_test.go:1314: making kafka feed factory
    helpers_test.go:1100: forcing enriched envelope for *changefeedccl.kafkaFeedFactory - EXPERIMENTAL CHANGEFEED FOR TABLE foo, TABLE bar WITH OPTIONS (on_error = 'pause', envelope = 'enriched', enriched_properties = 'source')
    helpers_test.go:1702: condition failed to evaluate within 45s: from helpers_test.go:1707: Expected status:paused but found status:running
--- FAIL: TestAlterChangefeedDropTargetAfterTableDrop/kafka (47.45s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedDropTargetAfterTableDrop.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51483

--- a/pkg/ccl/changefeedccl/enriched_source_provider.go
+++ b/pkg/ccl/changefeedccl/enriched_source_provider.go
@@ -540,6 +540,9 @@ func getDescriptors(
 		return nil
 	}
 	if err := execCfg.InternalDB.DescsTxn(ctx, f, isql.WithPriority(admissionpb.NormalPri)); err != nil {
+		if errors.Is(err, catalog.ErrDescriptorDropped) {
+			err = changefeedbase.WithTerminalError(err)
+		}
 		return nil, nil, nil, err
 	}
 	return tableDescriptor, dbDescriptor, schemaDescriptor, nil

